File: AdvertisementWebsite/BusinessLogic/Authorization/Permissions.cs
namespace BusinessLogic.Authorization;

public enum Permissions
{
    //User profile
    ViewProfileInfo = 0,
    EditProfileInfo,
    ChangePassword,

    //Advertisement bookmarks
    ViewAdvertisementBookmarks,
    BookmarkAdvertisement,
    EditAdvertisementBookmark,
    
    //Advertisement
    CreateAdvertisement,
    ViewOwnedAdvertisements,
    EditOwnedAdvertisement,
    DeleteOwnedAdvertisement,

    //Messages
    ViewMessages,
    SendMessage,

    //Advertisement notification subscriptions
    ViewAdvertisementNotificationSubscriptions,
    CreateAdvertisementNotificationSubscription,
    EditAdvertisementNotificationSubscriptions,
    DeleteAdvertisementNotificationSubscriptions,

    //Payment
    ViewPayments,
    MakePayment,

    //Manage all users
    ViewUsers
}

File: AdvertisementWebsite/advertisementwebsite.client/src/constants/api/Permissions.ts
export enum Permissions {
  //Manage own profile
  ViewOwnProfileInfo = 0,
  EditOwnProfileInfo,
  ChangeOwnPassword,
  //Manage own advertisement bookmarks
  ViewAdvertisementBookmarks,
  BookmarkAdvertisement,
  EditAdvertisementBookmark,
  //Manage own advertisements
  CreateOwnedAdvertisement,
  ViewOwnedAdvertisements,
  EditOwnedAdvertisement,
  DeleteOwnedAdvertisement,
  //Messages
  ViewMessages,
  SendMessage,
  //Manage own advertisement notification subscriptions
  ViewOwnedAdvertisementNotificationSubscriptions,
  CreateOwnedAdvertisementNotificationSubscription,
  EditOwnedAdvertisementNotificationSubscriptions,
  DeleteOwnedAdvertisementNotificationSubscriptions,
  //Payment
  ViewOwnPayments,
  MakePayment,
  //Manage categories
  CreateCategory,
  EditCategory,
  DeleteCategory,
  //Manage attributes
  ViewAllAttributes,
  CreateAttribute,
  EditAttribute,
  DeleteAttribute,
  //Manage users
  ViewAllUsers,
  CreateUser,
  EditAnyUser,
  DeleteAnyUser,
  //Manage roles
  ViewAllRoles,
  AddRole,
  EditRole,
  DeleteRole,
  //Manage permissions
  ViewAllPermissions,
  AddPermission,
  EditPermission,
  DeletePermission,
  //Manage advertisements
  CreateAdvertisement,
  ViewAllAdvertisements,
  EditAnyAdvertisement,
  DeleteAnyAdvertisement,
  //Manage advertisement notification subscriptions
  CreateAdvertisementNotificationSubscription,
  ViewAllAdvertisementNotificationSubscriptions,
  EditAnyAdvertisementNotificationSubscription,
  DeleteAnyAdvertisementNotificationSubscription,
  //Rule violation reports
  ViewRuleViolationReports,
  ResolveRuleViolationReport,
}